eservice-versions-archiver/src/main/scala/it/pagopa/interop/eserviceversionsarchiver/util/JobExecution.scala

```
@@ -0,0 +1,66 @@
+      _           <- archiveEserviceInternal(agreementId)
+    } yield ()
+
+  private def archiveEserviceInternal(agreementId: String): Future[Unit] = {
+    for {
+      maybeAgreement             <- readModelService
+        .findOne[PersistentAgreement](agreementsCollection, Filters.eq("data.id", agreementId))
+      (eServiceId, descriptorId) <- maybeAgreement
+        .toFuture(AgreementNotFound(agreementId))
+        .map(agreement => (agreement.eserviceId, agreement.descriptorId))
+      descriptorAndEserviceFilter = Filters.and(
+        Filters.eq("data.descriptorId", descriptorId),
+        Filters.eq("data.eserviceId", eServiceId)
+      )
+      relatingAgreements <- getAllAgreements(readModelService, descriptorAndEserviceFilter)
+      _                  <- {
+        val allArchived = relatingAgreements.forall(_.state == Archived)
+        if (allArchived) Future.unit
+        else Future.failed(NonArchivableDescriptorException(eServiceId, descriptorId))
+      }
+      maybeEservice <- readModelService.findOne[CatalogItem](eservicesCollection, Filters.eq("data.id", eServiceId))
+      eservice      <- maybeEservice.toFuture(EserviceNotFound(eServiceId))
+      descriptor    <- eservice.descriptors.find(_.id == descriptorId).toFuture(DescriptorNotFound(descriptorId))
+      _             <- {
+        descriptor.state match {
+          case Deprecated => Future.unit
+          case Suspended  =>
+            val newerDescriptorExists = eservice.descriptors
+              .exists(actualDescriptor =>
+                (actualDescriptor.state == Published || actualDescriptor.state == Suspended)
+                  && actualDescriptor.version.toInt > descriptor.version.toInt
+              )
+            if (newerDescriptorExists) Future.unit
+            else Future.failed(NonArchivableDescriptorException(eServiceId, descriptorId))
+          case _          => Future.failed(NonArchivableDescriptorException(eServiceId, descriptorId))
```

these conditions should not make the job fail, because they are not error/unexpected conditions.
Also in this case, the job should silently move on the next message